pkg/server/server.go

```diff
@@ -378,29 +378,31 @@ func (s *Server) SendVerificationLink(c *gin.Context, email string) {
 }
 
 // VerifyEmail marks a user email as verified
-func (s *Server) VerifyEmail(c *gin.Context) {
+func (s *Server) VerifyEmail(c *gin.Context, redirectURL string) {
 	jwtUser, exists := c.Get(types.JWTUserCredentialsKey)
 	if !exists {
 		s.redirectError(c, errors.ErrInvalidAccessToken)
+		c.Redirect(http.StatusUnauthorized, redirectURL)
 		return
 	}
 	jwtUserCredentials := jwtUser.(*models.UserCredentials)
 
-	updatedUserInfo, err := usermanager.UpdateUserDetails(s.userStore, jwtUserCredentials)
-	if err != nil {
-		s.redirect(c, err)
-	}
-
-	tgr := &jwtmanager.TokenGenerateRequest{
-		UserInfo:       updatedUserInfo,
-		AccessTokenExp: time.Minute * 10,
+	if jwtUserCredentials.UnverifiedEmail != nil {
+		*jwtUserCredentials.Email = jwtUserCredentials.GetUnverifiedEmail()
```

Comment:
Lets add a success log here that Email xxx@yyycom is verified successfully.

Reply:
Have added the log in the end before redirecting the user

Reply:
Ok, great